internal/copier/copier.go

```
@@ -5,35 +5,330 @@
+		}
+	}
+
+	return false
+}
+
+// checkPreservationRecursive checks if a path or any item within it (if it's a directory)
+// should be preserved based on hidden status or exclusion patterns.
+// It returns true if the path itself should be preserved OR if it's a directory
+// containing at least one item that should be preserved recursively OR if any parent
+// directory should be preserved.
+func checkPreservationRecursive(path string, excludePatterns []string) (bool, error) {
```

golangci-lint (1.64.8)

func `checkPreservationRecursive` is unused

(unused)